### PR TITLE
clearUpdatedConfiguration() API Android syntax and examples

### PR DESCRIPTION
**Pull request by spoorthipujariadobe**
Adds public documentation for the new public API `clearUpdatedConfiguration()` in Android

### CODE REVIEW

**Review by cdhoffmann — APPROVED** [review]

File: foundation-extensions/mobile-core/configuration/README.md [hunk 1]
```diff
@@ -263,7 +263,41 @@ ACPCore.UpdateConfiguration(config);
 {% endtab %}
 {% endtabs %}
 
-## Clearing programmatic updates to the configuration (iOS 3.x.x Only)
+## Clearing programmatic updates to the configuration (Android and iOS 3.x.x Only)
```
Comment (emdobrin) [inline foundation-extensions/mobile-core/configuration/README.md:266]: can we use the iOS (AEP 3.x) notation in these documents to be uniform with the other docs?

**Review by calebk1 — COMMENTED** [review]
Same comment as @emdobrin - we should be using the (AEP 3.x) syntax that we've been using everywhere else.

File: foundation-extensions/mobile-core/configuration/README.md [hunk 2]
```diff
@@ -263,7 +263,41 @@ ACPCore.UpdateConfiguration(config);
 {% endtab %}
 {% endtabs %}
 
-## Clearing programmatic updates to the configuration (iOS 3.x.x Only)
+## Clearing programmatic updates to the configuration (Android and iOS (AEP 3.x) Only)
```
Comment (emdobrin) [inline foundation-extensions/mobile-core/configuration/README.md:266]: @spoorthipujariadobe any strong reason to keep this on the extension home page or would the API reference be sufficient for this example?

File: foundation-extensions/mobile-core/configuration/configuration-api-reference.md [hunk 3]
```diff
@@ -465,9 +465,43 @@ MobileCore.configureWithFileInAssets("exampleJSONfile.json");
...
+`configureWithFileInPath(filePath)` -> `updateConfiguration(configMap)` -> `clearUpdatedConfiguration()`: In this example, you end up with the initial configuration set via `configureWithFileInPath(filePath)`
+
+`configureWithFileInAssets(fileName)` -> `updateConfiguration(configMap)` -> `clearUpdatedConfiguration()`: In this example, you end up with the initial configuration set via `configureWithFileInAssets(fileName)`
+
+`configureWithAppId(appId)` or `configureWithFileInPath(filePath)` or `configureWithFileInAssets(fileName)` -> `updateConfiguration(configMap)` -> `clearUpdatedConfiguration()` -> `updateConfiguration(configMap)`: In this example, the configuration will be the most recently updated configuration and will not have any keys from the first update unless they are included in the most recent update.
+
+`configureWithAppId(appId)` or `configureWithFileInPath(filePath)` or `configureWithFileInAssets(fileName)` -> `setPrivacyStatus(privacyStatus)` -> `clearUpdatedConfiguration()`: In this example, the configuration will have the initial `PrivacyStatus` set via `configureWithAppId(appId)` or `configureWithFileInPath(filePath)` or `configureWithFileInAssets(fileName)`
+
```
Comment (emdobrin) [inline foundation-extensions/mobile-core/configuration/configuration-api-reference.md:486]: Can we extract this in a more generic format outside of the language tabs and keep only the syntax / examples inside the tab or any other details specific to one platform only?

File: foundation-extensions/mobile-core/configuration/configuration-api-reference.md [hunk 4]
```diff
@@ -465,9 +465,43 @@ MobileCore.configureWithFileInAssets("exampleJSONfile.json");
...
 
-You can clear any programmatic updates made to the configuration via the `clearUpdatedConfiguration` API. This will clear programmatic updates to the configuration made via the `updateConfigurationWith(configDict:)` API. It will also clear any updates to the `PrivacyStatus` made via `setPrivacyStatus(_ status:)`. 
+{% tabs %}
+{% tab title="Android" %}
+
+You can clear any programmatic updates made to the configuration via the `clearUpdatedConfiguration` API. This will clear programmatic updates to configuration made via the `updateConfiguration(configMap)` API. It will also clear any updates to the `MobilePrivacyStatus` made via `setPrivacyStatus(privacyStatus)`.
+
+Here are some examples of scenarios:
+
+`configureWithAppId(appId)` -> `updateConfiguration(configMap)` -> `clearUpdatedConfiguration()`: In this example, you end up with the initial configuration set via `configureWithAppId(appId)`
```
Comment (emdobrin) [inline foundation-extensions/mobile-core/configuration/configuration-api-reference.md:477]: Let's we use a bullet list for these scenarios to improve readability

File: foundation-extensions/mobile-core/configuration/configuration-api-reference.md [hunk 5]
```diff
@@ -465,9 +465,43 @@ MobileCore.configureWithFileInAssets("exampleJSONfile.json");
 {% endtab %}
 {% endtabs %}
 
-## clearUpdatedConfiguration (iOS 3.x.x Only)
+## clearUpdatedConfiguration (Android and iOS (AEP 3.x) Only)
```
Comment (emdobrin) [inline foundation-extensions/mobile-core/configuration/configuration-api-reference.md:468]: Let's keep the APIs in alphabetical order.
Let's also move the version details under the header, you can use an info block: `This API is only available in Android and iOS (AEP 3.x).`

**Review by emdobrin — CHANGES_REQUESTED** [review]

File: foundation-extensions/mobile-core/configuration/README.md (hunk 2 above)
Comment (spoorthipujariadobe) [reply foundation-extensions/mobile-core/configuration/README.md:266]: The section above this one in the README has details on programmatically updating the configuration. I'm guessing that's why this section was added right below it. @cdhoffmann any other reason I'm missing?

File: foundation-extensions/mobile-core/configuration/configuration-api-reference.md (hunk 3 above)
Comment (emdobrin) [reply foundation-extensions/mobile-core/configuration/configuration-api-reference.md:486]: we can use just API name or sometimes we use the API name 1 (Android) / API name 2 (iOS) notation. would any of these work for your case?

Comment (spoorthipujariadobe) [reply foundation-extensions/mobile-core/configuration/configuration-api-reference.md:486]: The second format should work. Will update, thank you!

File: foundation-extensions/mobile-core/configuration/README.md [hunk 6]
```diff
@@ -263,47 +263,27 @@ ACPCore.UpdateConfiguration(config);
...
 
-**Syntax**
-```swift
-static func clearUpdatedConfiguration()
-```
+* `configureWithAppId(appId)`(Android)/`configureWith(appId:)`(iOS) or `configureWithFileInPath(filePath)`(Android)/ `configureWith(filePath:)`(iOS) or `configureWithFileInAssets(fileName)`(Android) -> `setPrivacyStatus(privacyStatus)`(Android)/ `setPrivacyStatus(_ status:)`(iOS) -> `clearUpdatedConfiguration()`: In this example, the configuration will have the initial `MobilePrivacyStatus`(Android)/ `PrivacyStatus`(iOS) set via `configureWithAppId(appId)`(Android)/`configureWith(appId:)`(iOS) or `configureWithFileInPath(filePath)`(Android)/ `configureWith(filePath:)`(iOS) or `configureWithFileInAssets(fileName)`(Android).
```
Comment (emdobrin) [inline foundation-extensions/mobile-core/configuration/README.md:284]: Let's also move the scenarios L274->L284 to the API reference page as they are in depth implementation details.

File: foundation-extensions/mobile-core/configuration/configuration-api-reference.md [hunk 7]
```diff
@@ -192,317 +160,367 @@ ACPCore.ConfigureWithAppID("1423ae38-8385-8963-8693-28375403491d");
...
 {% endtabs %}
 
-## updateConfiguration
-
-You can also update the configuration programmatically by passing configuration keys and values to override the existing configuration.
+## configureWithFileInAssets
 
 {% hint style="info" %}
-Keys that are not found on the current configuration are added when this method is followed. Null values are allowed and replace existing configuration values.
+This API is only available in Android.
```
Comment (emdobrin) [inline foundation-extensions/mobile-core/configuration/configuration-api-reference.md:166]: we can bundle the two info blocks together: `This API is only available in Android and was added in Android Core version 1.7.0.`

**Review by emdobrin — APPROVED** [review]
@spoorthipujari will you please address the comments below, looks good otherwise

**Pull request merged**